qcodes/dataset/plotting.py

```diff
@@ -627,6 +637,52 @@ def plot_on_a_plain_grid(x: np.ndarray,
     return ax, colorbar
 
 
+def _clip_nan_from_shaped_data(
+    x: np.ndarray, y: np.ndarray, z: np.ndarray
+) -> Tuple[np.ndarray, np.ndarray, np.ndarray]:
+    def _on_rectilinear_grid_except_nan(x_data: np.ndarray, y_data: np.ndarray) -> bool:
+        """
+        check that data is on a rectilinear grid. e.g. all points are the same as the  first
+        row and column with the exception of nans. Those represent points not yet measured.
+        """
+        x_row = x_data[:, 0:1]
+        y_row = y_data[0:1, :]
+        return (
+            np.nanmax(np.abs(x_data - x_row)) == 0
+            and np.nanmax(np.abs(y_data - y_row)) == 0
+        )
+
+    if _on_rectilinear_grid_except_nan(x, y):
+        """
+        clip any row or column where there are nans in the first row
+        or column. Since we fill from here we assume that means that no data
+        has been measured for this row
+        """
```

Comment:
not used to multiline string as a comment :)
```suggestion
        # clip any row or column where there are nans in the first row
        # or column. Since we fill from here we assume that means that no data
        # has been measured for this row
```